### ISSUE TITLE
clarification re identity parameters

### ISSUE THREAD

**Issue by mesamuels** [bcgsc/abyss#448]
What is the effective difference between the "p" parameter (minimum sequence identity of a bubble) and the pop bubble identity parameter? They are both set to default to 0.9. I tried changing the pop bubble to 0.95 as you suggested, it didn't have much effect. Do these two parameters need to be set the same to work correctly?

Vlad, fyi I found that increasing the k-mer length (from 96 to 104) helped significantly versus reducing it (from 96 to 88). After that it got slightly worse at least for total size stats. That may have to do with all the repetitive elements, which are better resolved with longer k-mers even if effective coverage goes down. I have to explore what happens to assembly of specific unique genes, where ideally each such gene (or part of a gene) should be in two contigs, coming from the two parental subgenomes of this allotetraploid plant. So far the results are exciting and excellent!

Thanks!
Mark

**Comment by vlad0x00** [comment]
Hey Mark,

`-p`, and `--identity` parameters are the same thing. Often command line programs have a short and long form for options, as in this case.

A change in k impacting assembly is expected. My guess was going lower, but of course you need to try multiple values to be sure!

**Comment by mesamuels** [comment]
Aha thanks Vlad, that is very helpful. I could imagine percent identity matching being used differently for creating and popping bubbles, although maybe that's my lack of understanding of the underlying algorithm. I wasn't sure how to put that command for the POPBUBBLE number into the batch script, but apparently I did it correctly. But putting the -p in the command line is simpler.

My cousin who is a retired engineer encouraged me to learn some programming, enough to be able to follow the execution of these kinds of complex programs, not in being able to understand the algorithms but to see how the programs are implementing. If I could, then probably in the source code it is clear that those two identity parameter commands do the same thing, and I wouldn't have to ask you. I'll try, although whether I can learn that much C++ will be seen.

If I understand the way the k-mer lengths work, if they're too short then there can be incorrect overlaps that need to be broken later. If they're too long, then the effective coverage of each k-mer goes down. In the 80-mer range, in principle that should be sufficient so longer k-mers wouldn't be helpful. But that's for a perfect unique genome sequence. When there are many thousands of very similar repeats, of sizes longer than the sequencing read length, then that must add a whole other level of complexity, making it harder to predict where the inflection point will be. In this case it seemed to be around 104bp. At 112bp the total assembled genome still got a bit larger, but the contigs got shorter. Not obvious which is more important.

For fun I'm trying to assemble the chloroplast, by requiring super high quality bases and a very large k-mer of 120. When I blasted contig files from earlier tries with a chloroplast gene sequence, there were many contig hits. I realize there might be some chloroplast sequence in nuclear chromosomes, but will be fun to see if among the outputs there is a (nearly) full length chloroplast.

Cheers!
Mark
